syntax = "proto3";

service DataNodeService {
  rpc ReplicateChunk(ReplicateChunkRequest) returns (ReplicateChunkResponse) {}
  rpc ListFiles(ListFilesRequest) returns (ListFilesResponse) {}
}

message ReplicateChunkRequest {
  string chunk_id = 1;
  bytes chunk_content = 2;
}

message ReplicateChunkResponse {
  bool success = 1;
}

message ListFilesRequest {
}

message ListFilesResponse {
    repeated string file_names = 1;
}